### ISSUE TITLE
Authoring extension example seems broken

### ISSUE THREAD

**Issue by andreasjakobik** [theia-ide/theia#1819]
Hello,

Following the "Authoring Theia Extensions" (http://www.theia-ide.org/doc/Authoring_Extensions.html), I was not able to successfully scaffold the project using the Theia extension generator.  Perhaps I'm missing something, but don't see what. See transcript below.

Thanks,
Andreas

```
$ yo theia-extension hello-world
HelloWorld
   create package.json
   create lerna.json
   create .gitignore
   create README.md
   create .vscode/launch.json
   create hello-world-extension/package.json
   create hello-world-extension/tsconfig.json
   create hello-world-extension/src/browser/hello-world-frontend-module.ts
   create hello-world-extension/src/browser/hello-world-contribution.ts
   create browser-app/package.json
   create electron-app/package.json
yarn install v1.6.0
info No lockfile found.
[1/4] Resolving packages...
warning workspace-aggregator-63696e29-598a-49d3-993e-6c7a20052d33 > hello-world-extension > @theia/core > @types/perfect-scrollbar@1.3.0: This is a stub types definition for perfect-scrollbar (https://github.com/noraesae/perfect-scrollbar). perfect-scrollbar provides its own type definitions, so you don't need @types/perfect-scrollbar installed!
warning workspace-aggregator-63696e29-598a-49d3-993e-6c7a20052d33 > browser-app > @theia/core > @types/perfect-scrollbar@1.3.0: This is a stub types definition for perfect-scrollbar (https://github.com/noraesae/perfect-scrollbar). perfect-scrollbar provides its own type definitions, so you don't need @types/perfect-scrollbar installed!
warning workspace-aggregator-63696e29-598a-49d3-993e-6c7a20052d33 > electron-app > @theia/core > @types/perfect-scrollbar@1.3.0: This is a stub types definition for perfect-scrollbar (https://github.com/noraesae/perfect-scrollbar). perfect-scrollbar provides its own type definitions, so you don't need @types/perfect-scrollbar installed!
warning workspace-aggregator-63696e29-598a-49d3-993e-6c7a20052d33 > browser-app > @theia/cli > @theia/application-manager > webpack-cli > jscodeshift > babel-preset-es2015@6.24.1: 🙌  Thanks for using Babel: we recommend using babel-preset-env now: please read babeljs.io/env to update! 
warning workspace-aggregator-63696e29-598a-49d3-993e-6c7a20052d33 > browser-app > @theia/cli > @theia/application-manager > webpack-cli > webpack-addons > jscodeshift > babel-preset-es2015@6.24.1: 🙌  Thanks for using Babel: we recommend using babel-preset-env now: please read babeljs.io/env to update! 
warning workspace-aggregator-63696e29-598a-49d3-993e-6c7a20052d33 > browser-app > @theia/cli > @theia/application-manager > webpack-cli > jscodeshift > nomnom@1.8.1: Package no longer supported. Contact support@npmjs.com for more info.
warning workspace-aggregator-63696e29-598a-49d3-993e-6c7a20052d33 > browser-app > @theia/cli > @theia/application-manager > webpack-cli > webpack-addons > jscodeshift > nomnom@1.8.1: Package no longer supported. Contact support@npmjs.com for more info.
warning workspace-aggregator-63696e29-598a-49d3-993e-6c7a20052d33 > hello-world-extension > @theia/core > electron > electron-download > nugget > progress-stream > through2 > xtend > object-keys@0.4.0: 
[2/4] Fetching packages...
warning vscode-base-languageclient@0.0.1-alpha.5: The engine "vscode" appears to be invalid.
info fsevents@1.2.3: The platform "linux" is incompatible with this module.
info "fsevents@1.2.3" is an optional dependency and failed compatibility check. Excluding it from installation.
[3/4] Linking dependencies...
warning "workspace-aggregator-63696e29-598a-49d3-993e-6c7a20052d33 > browser-app > @theia/cli > @theia/application-manager > font-awesome-webpack@0.0.5-beta.2" has unmet peer dependency "font-awesome@>=4.3.0".
[4/4] Building fresh packages...
success Saved lockfile.
$ lerna run prepare
lerna info version 2.4.0
hello-world-extension: > hello-world-extension@0.0.0 prepare /home/erajaik/work/learning/theia-hello-world-extension/hello-world-extension
hello-world-extension: > yarn run clean && yarn run build
hello-world-extension: yarn run v1.6.0
hello-world-extension: warning package.json: No license field
hello-world-extension: $ rimraf lib
hello-world-extension: Done in 0.13s.
hello-world-extension: yarn run v1.6.0
hello-world-extension: warning package.json: No license field
hello-world-extension: $ tsc
hello-world-extension: ../node_modules/@types/node/index.d.ts(2381,15): error TS2300: Duplicate identifier 'URL'.
hello-world-extension: ../node_modules/@types/node/index.d.ts(2399,15): error TS2300: Duplicate identifier 'URLSearchParams'.
hello-world-extension: ../node_modules/@types/node/index.d.ts(2417,14): error TS2661: Cannot export 'URL'. Only local declarations can be exported from a module.
hello-world-extension: ../node_modules/@types/node/index.d.ts(2417,19): error TS2661: Cannot export 'URLSearchParams'. Only local declarations can be exported from a module.
hello-world-extension: ../node_modules/typescript/lib/lib.dom.d.ts(13401,11): error TS2300: Duplicate identifier 'URL'.
hello-world-extension: ../node_modules/typescript/lib/lib.dom.d.ts(13417,13): error TS2300: Duplicate identifier 'URL'.
hello-world-extension: ../node_modules/typescript/lib/lib.dom.d.ts(13424,11): error TS2300: Duplicate identifier 'URLSearchParams'.
hello-world-extension: ../node_modules/typescript/lib/lib.dom.d.ts(13451,13): error TS2300: Duplicate identifier 'URLSearchParams'.
hello-world-extension: ../node_modules/typescript/lib/lib.dom.d.ts(15737,13): error TS2300: Duplicate identifier 'URL'.
hello-world-extension: ../node_modules/typescript/lib/lib.dom.d.ts(15738,13): error TS2300: Duplicate identifier 'URLSearchParams'.
hello-world-extension: info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
hello-world-extension: error Command failed with exit code 2.
hello-world-extension: npm ERR! code ELIFECYCLE
hello-world-extension: npm ERR! errno 1
hello-world-extension: npm ERR! hello-world-extension@0.0.0 prepare: `yarn run clean && yarn run build`
hello-world-extension: npm ERR! Exit status 1
hello-world-extension: npm ERR! 
hello-world-extension: npm ERR! Failed at the hello-world-extension@0.0.0 prepare script.
hello-world-extension: npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
hello-world-extension: npm ERR! A complete log of this run can be found in:
hello-world-extension: npm ERR!     /home/erajaik/.npm/_logs/2018-04-30T13_16_10_195Z-debug.log
lerna ERR! execute callback with error
lerna ERR! Error: Command failed: npm run prepare
lerna ERR! warning package.json: No license field
lerna ERR! warning package.json: No license field
lerna ERR! error Command failed with exit code 2.
lerna ERR! npm ERR! code ELIFECYCLE
lerna ERR! npm ERR! errno 1
lerna ERR! npm ERR! hello-world-extension@0.0.0 prepare: `yarn run clean && yarn run build`
lerna ERR! npm ERR! Exit status 1
lerna ERR! npm ERR! 
lerna ERR! npm ERR! Failed at the hello-world-extension@0.0.0 prepare script.
lerna ERR! npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
lerna ERR! 
lerna ERR! npm ERR! A complete log of this run can be found in:
lerna ERR! npm ERR!     /home/erajaik/.npm/_logs/2018-04-30T13_16_10_195Z-debug.log
lerna ERR! 
lerna ERR! 
lerna ERR! > hello-world-extension@0.0.0 prepare /home/erajaik/work/learning/theia-hello-world-extension/hello-world-extension
lerna ERR! > yarn run clean && yarn run build
lerna ERR! 
lerna ERR! yarn run v1.6.0
lerna ERR! $ rimraf lib
lerna ERR! Done in 0.13s.
lerna ERR! yarn run v1.6.0
lerna ERR! $ tsc
lerna ERR! ../node_modules/@types/node/index.d.ts(2381,15): error TS2300: Duplicate identifier 'URL'.
lerna ERR! ../node_modules/@types/node/index.d.ts(2399,15): error TS2300: Duplicate identifier 'URLSearchParams'.
lerna ERR! ../node_modules/@types/node/index.d.ts(2417,14): error TS2661: Cannot export 'URL'. Only local declarations can be exported from a module.
lerna ERR! ../node_modules/@types/node/index.d.ts(2417,19): error TS2661: Cannot export 'URLSearchParams'. Only local declarations can be exported from a module.
lerna ERR! ../node_modules/typescript/lib/lib.dom.d.ts(13401,11): error TS2300: Duplicate identifier 'URL'.
lerna ERR! ../node_modules/typescript/lib/lib.dom.d.ts(13417,13): error TS2300: Duplicate identifier 'URL'.
lerna ERR! ../node_modules/typescript/lib/lib.dom.d.ts(13424,11): error TS2300: Duplicate identifier 'URLSearchParams'.
lerna ERR! ../node_modules/typescript/lib/lib.dom.d.ts(13451,13): error TS2300: Duplicate identifier 'URLSearchParams'.
lerna ERR! ../node_modules/typescript/lib/lib.dom.d.ts(15737,13): error TS2300: Duplicate identifier 'URL'.
lerna ERR! ../node_modules/typescript/lib/lib.dom.d.ts(15738,13): error TS2300: Duplicate identifier 'URLSearchParams'.
lerna ERR! info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
lerna ERR! 
lerna ERR!     at Promise.all.then.arr (/home/erajaik/work/learning/theia-hello-world-extension/node_modules/execa/index.js:236:11)
lerna ERR!     at <anonymous>
(node:7754) UnhandledPromiseRejectionWarning: Error: Command failed: npm run prepare
warning package.json: No license field
warning package.json: No license field
error Command failed with exit code 2.
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! hello-world-extension@0.0.0 prepare: `yarn run clean && yarn run build`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the hello-world-extension@0.0.0 prepare script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/erajaik/.npm/_logs/2018-04-30T13_16_10_195Z-debug.log


> hello-world-extension@0.0.0 prepare /home/erajaik/work/learning/theia-hello-world-extension/hello-world-extension
> yarn run clean && yarn run build

yarn run v1.6.0
$ rimraf lib
Done in 0.13s.
yarn run v1.6.0
$ tsc
../node_modules/@types/node/index.d.ts(2381,15): error TS2300: Duplicate identifier 'URL'.
../node_modules/@types/node/index.d.ts(2399,15): error TS2300: Duplicate identifier 'URLSearchParams'.
../node_modules/@types/node/index.d.ts(2417,14): error TS2661: Cannot export 'URL'. Only local declarations can be exported from a module.
../node_modules/@types/node/index.d.ts(2417,19): error TS2661: Cannot export 'URLSearchParams'. Only local declarations can be exported from a module.
../node_modules/typescript/lib/lib.dom.d.ts(13401,11): error TS2300: Duplicate identifier 'URL'.
../node_modules/typescript/lib/lib.dom.d.ts(13417,13): error TS2300: Duplicate identifier 'URL'.
../node_modules/typescript/lib/lib.dom.d.ts(13424,11): error TS2300: Duplicate identifier 'URLSearchParams'.
../node_modules/typescript/lib/lib.dom.d.ts(13451,13): error TS2300: Duplicate identifier 'URLSearchParams'.
../node_modules/typescript/lib/lib.dom.d.ts(15737,13): error TS2300: Duplicate identifier 'URL'.
../node_modules/typescript/lib/lib.dom.d.ts(15738,13): error TS2300: Duplicate identifier 'URLSearchParams'.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

    at Promise.all.then.arr (/home/erajaik/work/learning/theia-hello-world-extension/node_modules/execa/index.js:236:11)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)
(node:7754) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 4)
(node:7754) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.

```

**Comment by marcdumais-work** [comment]
@andreasjakobik I do not know what the problem is, but I can confirm that I get the same issue, when I try.

**Comment by svenefftinge** [comment]
It is this : https://github.com/DefinitelyTyped/DefinitelyTyped/issues/25342

**Comment by svenefftinge** [comment]
A fix is to restrict the `@types/node` dependency to `^8.0.0`.

**Comment by andreasjakobik** [comment]
Great, thanks for quick answers!

I added Sven's suggestion in the ./hello-world-extension/package.json file:
```
...
"dependencies": {
    "@theia/core": "latest",
    "@types/node": "^8.0.0"
  },
...
```
After issuing `yarn` in root directory, everything goes fine.

Thanks,
Andreas

**Comment by marcdumais-work** [comment]
Closing this

**Comment by kittaakos** [comment]
@oweniano, let's open a new ticket next time.

> MSBUILD : error MSB4132: The tools version "2.0" is unrecognized. Available tools versions are "4.0".

You can check for a solution here:
 - https://github.com/theia-ide/yangster-electron/issues/24#issuecomment-372382721
 - https://stackoverflow.com/questions/14278417/cannot-install-node-modules-that-require-compilation-on-windows-7-x64-vs2012/22411007#22411007

Let us know if you're still having issues.
